### DISCUSSION TITLE
Can't assign my own ID to a vector in Milvus using Node.js sdk

### DISCUSSION THREAD

**Discussion by githycody** [milvus-io/milvus#30590] [Q&A and General discussion]
HI there, I am struggling to add my own SQL generated ID to the milvus db. Evertime I try, I am met with an error of the type:
error inserting data into Milvus database Error: Insert fail: some field does not exist for this collection in line. 0

I generate a id in a sqldb that gets autoIncremented, then retrieve this ID when I store text in it and pass in the ID as a function parameter in the milvus db, but it won't let me pass it in, I don't know what I am doing wrong. Please help! Also, everything worked when the ID parameter was set to auto increment in my Milvus db in terms of actually storing the vectors and assigning an id. I just want to be able to manually set one myself for each vector.

Here are the code snippets for both my vectordb.js file and app.js file:

**Vector db:**

const schema = [
    {
        name: 'ID',
        description: 'ID Field',
        data_type: DataType.Int64,
        is_primary_key: true,
        autoID: false,
    },
    {
        name: 'Vector',
        description: "Vector Embeddings",
        data_type: DataType.FloatVector,
        dim: 1536,
    },
];

await client.createCollection({
    collection_name,
    consistency_level: 'Strong',
    fields: schema,
});

// create a function that is going to hold the data and insert it into the milvus vector database
async function insertVectorIntoMilvus(id, vector) {
    const fields_data = [
        {
            ID: id, // Assuming `id` is a single value, not an array
            Vector: vector,
        },
    ];
   
**App.js:**

async function process_batches(batches) {
    for (let i = 0; i < batches.length; i++) {
        // let tokensBatches = 0;
        for (let j = 0; j < batches[i].length; j++) {
            //encode a chunk within a batch

            const chunkText = batches[i][j].pageContent;
            // console.log(`batch ${i + 1}, chunk ${j + 1}`, batches[i][j]);
            // tokensBatches += encodedChunk.length;
            if(chunkText != null){
                // insert chunk text inside of the database
                const chunkId = await insertChunksIntoDb(chunkText, pdfName);
                console.log("chunk ID:",chunkId);

                const embeddings = await create_embeddings(chunkText);
                console.log(`embeddings for batch ${i + 1}, and chunk ${j + 1}`, embeddings);

                if(embeddings && chunkId){
                    await insertVectorIntoMilvus(chunkId, embeddings);
                    console.log(`inserted embedding for batch ${i + 1} and chunk ${j + 1} into the Milvus Database`);
                }
            }
        }
        // console.log(`number of tokens for batch ${i + 1}:`, tokensBatches)
    }
}

**Comment by razataiab** [comment]
Hey, can you tell me how you resolved this?
